File: solutions/kotlin/difference-of-squares/1/src/main/kotlin/DifferenceOfSquares.kt
class Squares(
    private val n: Int
){
    fun sumOfSquares(): Int {
        var sum = 0
        for(i in 1..n) {
            sum += i * i
        }
        return sum
    }

    fun squareOfSum(): Int {
        var sum = 0
        for(i in 1..n) {
            sum += i
        }
        return sum * sum
    }

    fun difference() = squareOfSum() - sumOfSquares()
}

File: solutions/scala/difference-of-squares/1/src/main/scala/DifferenceOfSquares.scala
object DifferenceOfSquares {

  def sumOfSquares(n: Int): Int = {
    (1 to n)
        .map(x => x * x)
        .reduce(_ + _)
  }

  def squareOfSum(n: Int): Int = {
    val s = (1 to n).reduce(_ + _)
    s * s
  }

  def differenceOfSquares(n: Int): Int = {
    squareOfSum(n) - sumOfSquares(n)
  }
}
